modules/zemantaBidAdapter.js

```diff
@@ -0,0 +1,252 @@
+// jshint esversion: 6, es3: false, node: true
+'use strict';
+
+import {
+  registerBidder
+} from '../src/adapters/bidderFactory.js';
+import { NATIVE, BANNER } from '../src/mediaTypes.js';
+import * as utils from '../src/utils.js';
+import { ajax } from '../src/ajax.js';
+import { config } from '../src/config.js';
+
+const BIDDER_CODE = 'zemanta';
+const CURRENCY = 'USD';
+const NATIVE_ASSET_IDS = { 0: 'title', 2: 'icon', 3: 'image', 5: 'sponsoredBy', 4: 'body', 1: 'cta' };
+const NATIVE_PARAMS = {
+  title: { id: 0, name: 'title' },
+  icon: { id: 2, type: 1, name: 'img' },
+  image: { id: 3, type: 3, name: 'img' },
+  sponsoredBy: { id: 5, name: 'data', type: 1 },
+  body: { id: 4, name: 'data', type: 2 },
+  cta: { id: 1, type: 12, name: 'data' }
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [ NATIVE, BANNER ],
+  isBidRequestValid: (bid) => {
+    return !!(bid.nativeParams || bid.sizes);
+  },
+  buildRequests: (validBidRequests, bidderRequest) => {
+    const page = bidderRequest.refererInfo.referer;
+    const ua = navigator.userAgent;
+    const test = setOnAny(validBidRequests, 'params.test');
+    const publisher = setOnAny(validBidRequests, 'params.publisher');
+    const cur = CURRENCY;
+    const endpointUrl = config.getConfig('zemanta.bidderUrl');
```

Comment:
perhaps a better way of achieving this would be to utilize the **utils.buildUrl** function and have the user pass in an object that will then create a the endpoint. thoughts around this are you'll be able to verify it builds properly and matches a valid endpoint url for your bidder.

`pbjs.setConfig({
    zemanta: {
        bidderUrl: {
            protocol: 'https',
            host: 'bidderendpointurl',
            pathname: '/zenmax/ads',
        },
        usersyncUrl: {
            protocol: 'https',
            host: 'usersyncurl',
            pathname: '/zenmax/sync'
        }
    }
})`

Reply:
I would leave this as is since the publishers integrating with us will get different urls.

Reply:
What is the structure of the URL for bidder and sync endpoints? If the host is always the same you could have the user only define the path pointing to their url.  Also this should be updated to use the params of your bid to pass the URL to your bidder, not the config. You can see how other bidders like gamoshi are doing this. Do you want to include a backup endpoint in case there is none provided or it is incorrect?

The sync url should be passed by your endpoint in the response, as you know the endpoint url from the request. This is another point of failure that can be fixed.

Reply:
The host will change depending on the user and so could the path. The user will need to contact us before integrating so we see no issue providing them the full url. 

We do not support returning the usersync url in the response. The only place we can configure the usersync url then is in the config so we also moved the bidder url there to be consistent and not confuse the user by configuring the urls at two places.